NumLib/TimeStepping/Algorithms/FixedTimeStepping.h

```diff
@@ -59,32 +61,26 @@ class FixedTimeStepping : public ITimeStepAlgorithm
      */
     FixedTimeStepping(double t_initial, double t_end, const std::vector<double> &vec_all_dt);
 
-    /**
-     * @brief Create timestepper from the given configuration
-     *
-     * Currently this function only covers uniform timestep size.
-     */
-    static FixedTimeStepping* newInstance(BaseLib::ConfigTree const& config);
-
-    virtual ~FixedTimeStepping() {}
+    /// Create timestepper from the given configuration
+    static std::unique_ptr<ITimeStepAlgorithm> newInstance(BaseLib::ConfigTree const& config);
```

Comment:
Just a note not related to this PR. Sometimes there is a standalone method for a class `A`  `createA(ConfigTree, ...)`, sometimes there is a static member `A::newInstance(ConfigTree, ...)`. Having both "patterns" is potentially confusing.